src/backend/distributed/commands/table.c

```diff
@@ -138,6 +139,8 @@ PreprocessDropTableStmt(Node *node, const char *queryString,
 			continue;
 		}
 
+		LockParentShardResourcesForShardsOfPartition(relationId);
```

Comment:
We get the locks in preprocess because during the "process" and "postprocess" the table doesn't exist. But when we get the parent shard resource locks here, we have: `parent shard resource -> parent table -> partition table` lock order. However with UPDATE, we have `partition table -> parent shard resource` lock order so this creates a deadlock potential.